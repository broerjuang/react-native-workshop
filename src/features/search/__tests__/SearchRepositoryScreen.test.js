import renderer from 'react-test-renderer';
import React from 'react';
import {
  SearchRepositoryScreen,
  mapStateToProps,
} from '../screens/SearchRepositoryScreen';

describe('SearchRepositoryScreen Test', () => {
  it('should render SearchRepositoryScreen correctly', () => {
    let component = renderer.create(
      <SearchRepositoryScreen
        repos={['aji', 'deany', 'ari', 'jo', 'astrid']}
        navigation={{navigate: (s) => s}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should mapStateToProps correctly', () => {
    let state = {
      searchReducer: {
        repos: ['aji', 'deany', 'ari', 'jo', 'astrid'],
      },
    };
    let result = mapStateToProps(state);
    expect(result).toEqual(state.searchReducer);
  });
});
