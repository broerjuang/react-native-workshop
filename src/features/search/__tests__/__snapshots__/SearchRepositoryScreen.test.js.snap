// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRepositoryScreen Test should render SearchRepositoryScreen correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</RCTScrollView>
`;
